File: backend/src/main/java/com/certificategenerator/certificate/dto/CertificateRequest.java
package com.certificategenerator.certificate.dto;

import com.certificategenerator.certificate.CertificateStatus;
import com.certificategenerator.certificate.CertificateTemplate;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.time.LocalDate;

public record CertificateRequest(
        @NotBlank String recipientName,
        @NotBlank @Email String recipientEmail,
        @NotBlank String courseName,
        @NotNull @Positive Integer workloadHours,
        @NotNull LocalDate completionDate,
        @NotNull LocalDate issueDate,
        @NotBlank String instructorName,
        @NotNull CertificateTemplate template,
        // Optional: omitted on create defaults to DRAFT, per design.md — not listed among the
        // spec's required fields, unlike every other field here.
        CertificateStatus status) {}

File: frontend/src/app/features/certificates/data/certificate-request.ts
import { CertificateStatus, CertificateTemplate } from "./certificate-page-response";

/** Matches backend/.../certificate/dto/CertificateRequest.java (docs/api-reference.md). `status` is optional — omitted on create, defaults server-side to DRAFT. */
export interface CertificateRequest {
  recipientName: string;
  recipientEmail: string;
  courseName: string;
  workloadHours: number;
  completionDate: string;
  issueDate: string;
  instructorName: string;
  template: CertificateTemplate;
  status?: CertificateStatus;
}
